packages/vite/src/node/plugins/asset.ts

```diff
@@ -36,6 +42,92 @@ export function registerCustomMime(): void {
   mrmime.mimes['eot'] = 'application/vnd.ms-fontobject'
 }
 
+export function renderAssetUrl(
+  ctx: PluginContext,
+  config: ResolvedConfig,
+  chunk: RenderedChunk,
+  opts: NormalizedOutputOptions,
+  code: string,
+  assetFileName: string,
+  hostType: 'js' | 'css' | 'html',
```

Comment:
I think we can remove the host type if we only call it for `'js'`, for this PR maybe better to avoid touching the replace in CSS function here: https://github.com/vitejs/vite/pull/9507/files#r947125650
But this is a refactoring we could do in 3.1 if we think it helps.